src/victory-util/add-events.js

```diff
@@ -26,15 +26,25 @@ export default (WrappedComponent) => {
       const { sharedEvents } = props;
       const components = WrappedComponent.expectedComponents;
       this.componentEvents = Events.getComponentEvents(props, components);
-      this.baseProps = isFunction(WrappedComponent.getBaseProps) ?
-        WrappedComponent.getBaseProps(props) : {};
-      this.dataKeys = Object.keys(this.baseProps).filter((key) => key !== "parent");
       this.getSharedEventState = sharedEvents && isFunction(sharedEvents.getEventState) ?
         sharedEvents.getEventState : () => undefined;
+      this.baseProps = this.getBaseProps(props);
+      this.dataKeys = Object.keys(this.baseProps).filter((key) => key !== "parent");
```

Comment:
I'd probably use `reject` here ("Reject parent keys" rather than "use keys that aren't parent"). It reads easier to me. Not sure if you're into using both `reject` and `filter` though, or if you're trying to minimize dependency on `lodash`.